lib/src/builder/generation/impl_generation.dart

```diff
@@ -247,6 +247,9 @@ class ImplGenerator {
             ..writeln()
             ..writeln('  @override')
             ..writeln('  set state(Map value) {')
+            ..writeln('    assert(value is JsBackedMap, ')
+            ..writeln('      \'Component2.state may only be set to a JsBackedMap, \'')
+            ..writeln('        \'and must not be set outside of the react-dart internals.\');')
```

Comment:
Can we make this message for helpful to the consumer by instructing them to use the `initializeState` method instead?

Reply:
Yep, good call. Because it's the general state setter I updated the message to also mention setState. Let me know if the new message could be improved as well!